cardinal/world_context.go

```diff
@@ -102,6 +114,10 @@ func (ctx *worldContext) Logger() *zerolog.Logger {
 	return ctx.logger
 }
 
+func (ctx *worldContext) Rand() *rand.Rand {
```

Comment:
We need to add a conditional check here to ensure that this is only used with `newWorldContextForTick` similar to how there is a read-only check. 

This is to prevent people from using this via `NewWorldContext` and `NewReadOnlyWorldContext` (used by queries) that will produce invalid behaviors for the timestamp since it's not executing within a tick.